sklearn/preprocessing/data.py

```diff
@@ -788,7 +788,7 @@ def transform(self, X, y=None):
             warnings.warn(DEPRECATION_MSG_1D, DeprecationWarning)
 
         if sparse.issparse(X):
-            if X.shape[0] == 1:
+            if X.shape[1] != self.scale_.shape[0]:
```

Comment:
oooo, so there's no need to consider the case where there is an array of number (shape of (1, n)) being fit-transformed? MaxAbsScaler only does scaling for (m, n) matrix for each column, where m > 1?